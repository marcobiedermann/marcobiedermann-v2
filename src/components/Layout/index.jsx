import PropTypes from 'prop-types';
import React from 'react';
import * as routes from '../../constants/routes';
import { now } from '../../utils/date';
import Footer from '../Footer';
import Grid from '../Grid';
import Header from '../Header';
import Main from '../Main';
import Navigation from '../Navigation';

const Layout = props => {
  const { children, ...otherProps } = props;

  return (
    <div {...otherProps}>
      <Header>
        <Grid>
          <Navigation routes={[routes.INDEX, routes.PROJECTS, routes.SEARCH, routes.IMPRINT]} />
        </Grid>
      </Header>
      <Main>{children}</Main>
      <Footer>
        <Grid>
          Â© {now.getFullYear()}, Built with <a href="https://www.gatsbyjs.org">Gatsby</a>
        </Grid>
      </Footer>
    </div>
  );
};

Layout.propTypes = {
  children: PropTypes.node,
};

Layout.defaultProps = {
  children: null,
};

export default Layout;
