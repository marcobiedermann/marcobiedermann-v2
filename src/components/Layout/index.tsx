import classNames from 'classnames';
import { Link } from 'gatsby';
import React, { FC } from 'react';
import { useLayout } from '../../hooks/layout';
import Footer from '../Footer';
import Grid from '../Grid';
import Header from '../Header';
import Main from '../Main';
import Navigation from '../Navigation';
import styles from './style.module.css';

export interface LayoutProps {
  className?: string;
}

export const Layout: FC<LayoutProps> = (props) => {
  const { children, className, ...otherProps } = props;
  const { allContentfulMenu } = useLayout();

  const { edges: menus } = allContentfulMenu;

  const mappedMenus = menus.map((menu) => menu.node);
  const mainMenu = mappedMenus.find((menu) => menu.slug === 'main');
  const footerMenu = mappedMenus.find((menu) => menu.slug === 'footer');

  return (
    <div className={classNames(className, styles.layout)} {...otherProps}>
      <Header>
        <Link to="/" style={{ textDecoration: 'none' }}>
          marcobiedermann
        </Link>
        {mainMenu && <Navigation inline routes={mainMenu.routes} />}
      </Header>
      <Main>{children}</Main>
      <Footer>
        <Grid justify="space-between">
          <p>Handcrafted with love in Germany</p>
          {/* <p>Happy {format(DATE_NOW, 'iiii')}</p> */}
          {footerMenu && <Navigation inline routes={footerMenu.routes} />}
        </Grid>
      </Footer>
    </div>
  );
};

export default Layout;
