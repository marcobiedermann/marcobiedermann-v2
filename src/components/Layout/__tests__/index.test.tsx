/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import { useStaticQuery } from 'gatsby';
import React from 'react';
import Layout from '..';
import route from '../../Route/__fixtures__';

const mockedUseStaticQuery = useStaticQuery as jest.Mock;

jest.mock('../../../constants/date', () => ({
  DATE_NOW: new Date('2000-01-01'),
}));

describe('Layout component', () => {
  beforeEach(() => {
    mockedUseStaticQuery.mockImplementationOnce(() => ({
      allContentfulMenu: {
        edges: [
          {
            node: {
              slug: 'main',
              routes: [route],
            },
          },
          {
            node: {
              slug: 'footer',
              routes: [route],
            },
          },
        ],
      },
    }));
  });

  it('should render correctly', () => {
    const { asFragment } = render(<Layout />);

    expect(asFragment()).toMatchSnapshot();
  });
});
