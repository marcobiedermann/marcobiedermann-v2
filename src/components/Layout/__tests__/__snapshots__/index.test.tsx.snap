// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <header
      class="header"
    >
      <a
        href="/"
        style="text-decoration: none;"
      >
        marcobiedermann
      </a>
      <nav
        class="navigation"
      >
        <ul
          class="navigation__list navigation__listInline"
        >
          <li>
            <a
              class="navigation__link"
              href="/path/to/page"
              id="b6c3ceb6-d89f-4a03-9e96-77768f278909"
            >
              Page name
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="main"
    />
    <footer
      class="footer"
    >
      <div
        class="grid alignment alignmentJustifySpaceBetween"
      >
        <p>
          Handcrafted with love in Germany
        </p>
        <nav
          class="navigation"
        >
          <ul
            class="navigation__list navigation__listInline"
          >
            <li>
              <a
                class="navigation__link"
                href="/path/to/page"
                id="b6c3ceb6-d89f-4a03-9e96-77768f278909"
              >
                Page name
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
